.github/workflows/tests.yaml

```diff
@@ -23,19 +23,26 @@ jobs:
         # At present, these images are newer than "latest". We use them to test
         # against more recent tooling versions.
         # https://github.com/actions/runner-images
-        os: [ubuntu-22.04, macos-12]
+        os: [ubuntu-22.04, macos-14]
         build_mode: [fastbuild, opt]
-        # The clang-tidy config doesn't work on macos (missing `truncate`).
         include:
+          # The clang-tidy config doesn't work on macos (missing `truncate`).
           - os: 'ubuntu-22.04'
             build_mode: 'clang-tidy'
+          # On push (post-submit), also run on `macos-12` to get Intel macOS
+          # coverage. This is less and less critical so we skip it on the PR and
+          # merge queue path by switching to a redundant OS.
+          - os: ${{ github.event_name == 'push' && 'macos-12' || 'macos-14' }}
```

Comment:
It's a bit harder to do than that because you can't put JSON directly into an expression, but you can parse a string into JSON, so I've done that. It does seem simpler, even if it breaks VSCode's YAML highlighting...